Sources/Kevin/Core/Custom/KevinTextField.swift

```diff
@@ -0,0 +1,150 @@
+        textField.leftAnchor.constraint(equalTo: self.leftAnchor).isActive = true
+        textField.rightAnchor.constraint(equalTo: self.rightAnchor).isActive = true
+        textField.topAnchor.constraint(equalTo: self.topAnchor).isActive = true
+        textFieldBottomConstraint = textField.bottomAnchor.constraint(equalTo: self.bottomAnchor, constant: 0)
+        textFieldBottomConstraint?.isActive = true
+        textField.addTarget(self, action: #selector(textFieldTextChanged), for: .editingChanged)
+        
+        addSubview(errorLabel)
+        errorLabel.layer.zPosition = 1
+        errorLabel.text = ""
+        errorLabel.numberOfLines = 0
+        errorLabel.textColor = borderColor
+        errorLabel.alpha = 0
+        errorLabel.translatesAutoresizingMaskIntoConstraints = false
+        errorLabel.leftAnchor.constraint(equalTo: self.leftAnchor).isActive = true
+        errorLabel.rightAnchor.constraint(equalTo: self.rightAnchor).isActive = true
+        errorLabel.bottomAnchor.constraint(equalTo: self.bottomAnchor).isActive = true
+        labelTopConstraint = errorLabel.topAnchor.constraint(equalTo: textField.bottomAnchor, constant: 8)
+        labelTopConstraint?.isActive = false
+    }
+    
+    func setError(_ errorMessage: String?) {
+        if let errorMessage = errorMessage {
+            showError(errorMessage)
+        } else {
+            hideError()
+        }
+    }
+    
+    private func showError(_ errorMessage: String) {
+        if preErrorTextFieldBackgroundColor == nil {
+            preErrorTextFieldBackgroundColor = textField.backgroundColor
+            preErrorTextFieldBorderColor = textField.layer.borderColor
+        }
+        
+        errorLabel.text = errorMessage
+        textFieldBottomConstraint?.isActive = false
+        labelTopConstraint?.isActive = true
+        
+        UIView.animate(withDuration: animationDuration) {
+            self.textField.layer.borderColor = self.borderColor.cgColor
+
+            self.errorLabel.alpha = 1
+            self.superview?.layoutIfNeeded()
+        }
+    }
+    
+    private func hideError() {
+        
+        labelTopConstraint?.isActive = false
+        textFieldBottomConstraint?.isActive = true
+        
+        UIView.animate(withDuration: animationDuration) {
+            if self.preErrorTextFieldBackgroundColor != nil {
+                self.textField.backgroundColor = self.preErrorTextFieldBackgroundColor
+                self.textField.layer.borderColor = self.preErrorTextFieldBorderColor
+            }
+            
+            self.errorLabel.alpha = 0
+            self.superview?.layoutIfNeeded()
+        } completion: { _ in
+            self.errorLabel.text = ""
+            self.preErrorTextFieldBackgroundColor = nil
+            self.preErrorTextFieldBorderColor = nil
+        }
+    }
+    
+    @objc fileprivate func textFieldTextChanged() {
+        hideError()
+    }
+}
+
+class KevinTextField: UITextField {
```

Comment:
move to the separate file, but I think these 2 classes can be merged into a single one like it is done [here](https://github.com/edgar-zigis/CocoaTextField)